tests/webhook_controller_test.go

```diff
@@ -0,0 +1,89 @@
+package tests
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	sspv1beta2 "kubevirt.io/ssp-operator/api/v1beta2"
+	"kubevirt.io/ssp-operator/controllers"
+	"kubevirt.io/ssp-operator/tests/env"
+)
+
+var _ = Describe("Webhook controller", func() {
+	var webhook *admissionv1.ValidatingWebhookConfiguration
+
+	BeforeEach(func() {
+		webhook = &admissionv1.ValidatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "ssp.kubevirt.io-test-webhook-",
+				Labels: map[string]string{
+					controllers.OlmNameLabel: controllers.OlmNameLabelValue,
+				},
+			},
+			Webhooks: []admissionv1.ValidatingWebhook{{
+				Name: "webhook",
+				ClientConfig: admissionv1.WebhookClientConfig{
+					Service: &admissionv1.ServiceReference{
+						Namespace: "",
+						Name:      "",
+						Path:      nil,
+						Port:      nil,
+					},
+				},
+				Rules: []admissionv1.RuleWithOperations{{
+					// Using "Delete" so it does not conflict with existing SSP webhook
+					Operations: []admissionv1.OperationType{admissionv1.Delete},
+					Rule: admissionv1.Rule{
+						APIGroups:   []string{sspv1beta2.GroupVersion.Group},
+						APIVersions: []string{sspv1beta2.GroupVersion.Version},
+						Resources:   []string{"ssps"},
+					},
+				}},
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"ssp-webhook-test-label": "ssp-webhook-test-label-vale",
+					},
+				},
+				SideEffects:             ptr.To(admissionv1.SideEffectClassNone),
+				AdmissionReviewVersions: []string{"v1"},
+			}},
+		}
+	})
+
+	It("[test_id:TODO] should delete namespaceSelector from webhook configuration with OLM label", func() {
+		Expect(apiClient.Create(ctx, webhook)).To(Succeed())
+		DeferCleanup(func() {
+			Expect(apiClient.Delete(ctx, webhook)).To(Succeed())
+		})
```

Comment:
As far as I understand, the intention here is that these `DeferCleanup` function calls are executed before the global `AfterEach`:

https://github.com/kubevirt/ssp-operator/blob/eb60b158ff1d5914f6ec3cfdd32d51b5e884f4e3/tests/cleanup_test.go#L29-L31

Could you verify that this is what actually happens?  As mentioned before, these calls have an issue which makes them to be executed after the global `AfterEach`:  https://github.com/onsi/ginkgo/issues/1284#issuecomment-1756314394

Reply:
That issue is probably not relevant to this specific test file, because it does not use `AfterEach()`.

I will read `ginkgo` source to be sure how the ordering is done.

Btw, what do you mean by "global `AfterEach()`" ? Do you mean the `AfterSuite()` function:
https://github.com/kubevirt/ssp-operator/blob/b1770163ba65c1327926a47de79c6815c3a75eb6/tests/tests_suite_test.go#L419-L421

Reply:
Sorry for the noise, SSP does not have such `AfterEach` executed after each test case as the `KubeVirt` testsuite. So in this case, this shouldn't be a problem.

Reply:
In this case it is most likely not a problem, but a global awareness of the caveats of `DeferCleanup` is certainly helpful.